### PR TITLE
GEOMESA-659 Implement raster chunking functionality in Geomesa tools

### PR DESCRIPTION
**Pull request by wjp22903**

### CODE REVIEW

**Comment by aannex-ccri** [comment]
please update the readme in geomesa-tools to explain the ingestRaster command.

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    new File(outDir).mkdir
+
+    val indexes =
+      (for {
+        i <- 0 to stepsX
+        j <- 0 to stepsY
+      } yield (i, j)).par
+    indexes.tasksupport = new ForkJoinTaskSupport(new scala.concurrent.forkjoin.ForkJoinPool(parLevel))
+
+    indexes.foreach{ case (i, j) =>
```
Comment (jnh5y) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala]: (picky) Add a space between the foreach and the {.

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+   * @param size  Desired size limit in bytes
+   */
+  def chunkRasterLocal(file: String, fileType: String, outDir: String, size: Double) {
+    val rasterReader = getReader(new File(file), fileType)
+    val rasterGrid: GridCoverage2D = rasterReader.read(null)
+    val envelope = rasterGrid.getEnvelope2D
+    val bbox = BoundingBox(envelope.getMinX, envelope.getMaxX, envelope.getMinY, envelope.getMaxY)
+    val (gridStepX, gridStepY) = RasterChunking.getSteps(rasterGrid)
+    val chunkPrec = RasterChunking.GetGeoHashPrecisionBySize(rasterGrid, size)
+    val llGh = GeoHash(envelope.getMinX, envelope.getMinY, chunkPrec)
```
Comment (jnh5y) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala]: We should add code to help deal with input with inputs which may exceed the 'whole world' bounding box.

**Comment by jnh5y** [comment]
Is there any code to delete the temporary chunking directory?

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    }
+    Math.min(Math.max(calCoor, minCoor), maxCoor)
+  }
+}
+
+object RasterChunking extends Logging {
+  val BOUND_MIN_X = -180.0
+  val BOUND_MIN_Y = -90.0
+  val BOUND_MAX_X = 180.0
+  val BOUND_MAX_Y = 89.999999
```
Comment (aannex-ccri) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala]: shouldn't this be 90.0?

Comment (jnh5y) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala]: Yes....   But in order for that to work, GeoHash(180.0, 90.0) needs to do something sensible.  (I told Jianping to do this.)

Comment (aannex-ccri) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala]: gotcha

**Comment by ronq** [comment]
Merged into f_wcs_4 as commit 03dddf31e8418fffcf24c9d38c2e46c0cec29da1

**Pull request closed without merging**